src/EnergyPlus/WeatherManager.cc

```diff
@@ -2045,6 +2045,29 @@ namespace WeatherManager {
                     }
                 }
             }
+
+            // at the beginning of each day find the min/max weather used for DOAS sizing
+            if (state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).KindOfEnvrn ==
+                    DataGlobalConstants::KindOfSim::RunPeriodDesign ||
+                state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay) {
+                for (int i = 0; i < state.dataWeatherManager->TodayOutDryBulbTemp.size(); ++i) {
+                    if (state.dataWeatherManager->TodayOutDryBulbTemp[i] >
+                        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOATSizing) {
+                        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOATSizing =
+                            state.dataWeatherManager->TodayOutDryBulbTemp[i];
+                        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOADPSizing =
+                            state.dataWeatherManager->TodayOutDewPointTemp[i];
+                    }
+                    if (state.dataWeatherManager->TodayOutDryBulbTemp[i] <
+                        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOATSizing) {
+                        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOATSizing =
+                            state.dataWeatherManager->TodayOutDryBulbTemp[i];
+                        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOADPSizing =
+                            state.dataWeatherManager->TodayOutDewPointTemp[i];
+                    }
+                }
+            }
```

Comment:
For reference (pun intended), consider the readability of these two:

```
if (state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).KindOfEnvrn ==
        DataGlobalConstants::KindOfSim::RunPeriodDesign ||
    state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay) {
  for (int i = 0; i < state.dataWeatherManager->TodayOutDryBulbTemp.size(); ++i) {
    if (state.dataWeatherManager->TodayOutDryBulbTemp[i] >
        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOATSizing) {
      state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOATSizing =
          state.dataWeatherManager->TodayOutDryBulbTemp[i];
      state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).maxCoolingOADPSizing =
          state.dataWeatherManager->TodayOutDewPointTemp[i];
    }
    if (state.dataWeatherManager->TodayOutDryBulbTemp[i] <
        state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOATSizing) {
      state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOATSizing =
          state.dataWeatherManager->TodayOutDryBulbTemp[i];
      state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).minHeatingOADPSizing =
          state.dataWeatherManager->TodayOutDewPointTemp[i];
    }
  }
}
```

vs

```
auto &thisEnvrn = state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn);
if (thisEnvrn.KindOfEnvrn == DataGlobalConstants::KindOfSim::RunPeriodDesign || thisEnvrn.KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay) {
  for (int i = 0; i < state.dataWeatherManager->TodayOutDryBulbTemp.size(); ++i) {
    auto &thisDB = state.dataWeatherManager->TodayOutDryBulbTemp[i];
    auto &thisDP = state.dataWeatherManager->TodayOutDewPointTemp[i];
    if (thisDB > thisEnvrn.maxCoolingOATSizing) {
      thisEnvrn.maxCoolingOATSizing = thisDB;
      thisEnvrn.maxCoolingOADPSizing = thisDP;
    }
    if (thisDB < thisEnvrn.minHeatingOATSizing) {
      thisEnvrn.minHeatingOATSizing = thisDB;
      thisEnvrn.minHeatingOADPSizing = thisDP;
    }
  }
}
```

Just by adding a few reference variables, the logic/algorithmic part of the code becomes instantly and significantly more readable.

Reply:
:rage: @Myoldmopar, you and your `auto &` for scalars! Just use `Real64` for `thisDB` and `thisDP`. Telling the compiler that you want to access these by reference just gives the compiler two more addresses to track in this function.  

PS Weather manager should be refactored to use a single 2D array (or even a 1D array) of `structs` rather than a separate 2D array for each quantity. That would also simplify these kinds of loops. :pushpin:

Reply:
Yep! Sorry about that. I'm so accustomed to creating references to the objects, it just gets into my muscle memory.

Reply:
The last tight block does look much better.

Reply:
@rraustad (and everyone). The general "rule" is:

- `auto &` (or `auto const &`) for classes. I hate `auto` in general, but using it to mean "`class` whose name I can't quite remember right now and is probably too long anyway" makes it slightly less terrible.
- The actual type for scalars and strings with no `&`. `int`, `Real64`, `bool` are all about as long as `auto` and not hard to remember. I would also use `std::string`, `std::string &` and `std::string_view` explicitly because they have different copy semantics and I am not sure the compiler can figure out which to use on its own.
- No `auto &` for arrays themselves.